x-pack/plugins/siem/public/components/page/hosts/first_last_seen_host/index.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiLoadingSpinner } from '@elastic/eui';
+import moment from 'moment';
+import React from 'react';
+import { ApolloConsumer } from 'react-apollo';
+import { pure } from 'recompose';
+
+import { useFirstLastSeenHostQuery } from '../../../../containers/hosts/first_last_seen';
+import { getEmptyTagValue } from '../../../empty_value';
+import { PreferenceFormattedDate } from '../../../formatted_date';
+import { LocalizedDateTooltip } from '../../../localized_date_tooltip';
+
+export type FirstLastSeenHostType = 'first-seen' | 'last-seen';
+
+export const FirstLastSeenHost = pure<{ hostname: string; type: FirstLastSeenHostType }>(
+  ({ hostname, type }) => {
+    return (
+      <ApolloConsumer>
+        {client => {
+          const { loading, firstSeen, lastSeen } = useFirstLastSeenHostQuery(
+            hostname,
+            'default',
+            client
+          );
+          return (
+            <>
+              {loading && <EuiLoadingSpinner size="m" />}
+              {!loading &&
+              type === 'first-seen' &&
+              firstSeen != null &&
+              new Date(firstSeen).toString() === 'Invalid Date'
+                ? firstSeen
+                : !loading &&
```

Comment:
This extra `!loading` is not needed. You already tested for this boolean condition before this nested conditional is reached.

Reply:
This is actually needed as the `!loading` is the start of the second half of the ternary. See [my comment](https://github.com/elastic/kibana/pull/34831#discussion_r274103224) above about simplifying this block.